index.js

```diff
@@ -0,0 +1,39 @@
+'use strict'
+
+const fp = require('fastify-plugin')
+
+async function plugin (app, options) {
+  const clientsRole = options.mtlsClientsRole ?? null
+  const commonNameDomain = `.${options.mtlsCommonNameDomain}`
+
+  app.decorateRequest('createMtlsSession', async function () {
+    const certificate = this.raw.socket.getPeerCertificate(false)
```

Comment:
You should throw a proper error in case this is not available (for example in case of `inject()`